File: lib/ui/screens/bills_and_subscriptions_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import '../../core/theme/app_colors.dart';
import '../../core/theme/design_tokens.dart';
import '../../core/util/currency_formatter.dart';
import '../../domain/models/domain_models.dart';
import '../viewmodel/pfms_provider.dart';

class BillsAndSubscriptionsScreen extends ConsumerWidget {
  final VoidCallback onNavigateBack;

  const BillsAndSubscriptionsScreen({super.key, required this.onNavigateBack});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final controller = PfmsController(ref);
    final bills = ref.watch(billsStreamProvider).value ?? [];
    final accounts = ref.watch(accountsStreamProvider).value ?? [];
    final isArabic = ref.watch(isArabicProvider);

    return Scaffold(
      appBar: AppBar(
        title: Text(isArabic ? "الفواتير والاشتراكات" : "Bills & Subscriptions"),
        leading: IconButton(icon: const Icon(Icons.arrow_back), onPressed: onNavigateBack),
      ),
      floatingActionButton: FloatingActionButton(
        backgroundColor: AppColors.tealAccentLight,
        onPressed: () => controller.openBottomSheet(QuickActionSheetType.BILL),
        child: const Icon(Icons.add, color: Colors.black),
      ),
      body: ListView.builder(
        padding: const EdgeInsets.all(DesignTokens.spaceMedium),
        itemCount: bills.length,
        itemBuilder: (context, index) {
          final bill = bills[index];
          final isPaid = bill.status == BillStatus.PAID;

          return Card(
            color: AppColors.slateDarkCard,
            margin: const EdgeInsets.only(bottom: DesignTokens.spaceSmall),
            child: ListTile(
              leading: CircleAvatar(
                backgroundColor: isPaid ? AppColors.greenIncome.withOpacity(0.2) : AppColors.goldAccent.withOpacity(0.2),
                child: Icon(Icons.receipt_long, color: isPaid ? AppColors.greenIncome : AppColors.goldAccent),
              ),
              title: Text(bill.title, style: const TextStyle(fontWeight: FontWeight.bold, color: AppColors.textPrimaryDark)),
              subtitle: Text("${bill.frequency} • ${CurrencyFormatter.format(bill.amount, bill.currency)}", style: const TextStyle(color: AppColors.textMutedDark)),
              trailing: isPaid
                  ? Text(isArabic ? "مدفوعة" : "Paid", style: const TextStyle(color: AppColors.greenIncome, fontWeight: FontWeight.bold))
                  : ElevatedButton(
                      style: ElevatedButton.styleFrom(backgroundColor: AppColors.tealAccent),
                      onPressed: () {
                        if (accounts.isNotEmpty) {
                          controller.payBill(bill, accounts.first.id);
                        }
                      },
                      child: Text(isArabic ? "دفع" : "Pay", style: const TextStyle(color: Colors.black)),
                    ),
            ),
          );
        },
      ),
    );
  }
}

File: app/src/main/java/com/example/ui/screens/BillsAndSubscriptionsScreen.kt
package com.example.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.example.core.theme.DesignTokens
import com.example.core.util.CurrencyFormatter
import com.example.domain.model.BillStatus
import com.example.ui.theme.GreenIncome
import com.example.ui.viewmodel.PfmsViewModel
import com.example.ui.viewmodel.QuickActionSheetType
import java.text.SimpleDateFormat
import java.util.*

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun BillsAndSubscriptionsScreen(
    viewModel: PfmsViewModel,
    onNavigateBack: () -> Unit
) {
    val bills by viewModel.bills.collectAsState()
    val accounts by viewModel.accounts.collectAsState()
    val isArabic by viewModel.isArabic.collectAsState()

    val dateFormat = SimpleDateFormat("dd MMM yyyy", Locale.US)

    Scaffold(
        topBar = {
            TopAppBar(
                title = {
                    Text(
                        text = if (isArabic) "الفواتير والاشتراكات" else "Bills & Subscriptions",
                        fontWeight = FontWeight.Bold
                    )
                },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = if (isArabic) "رجوع" else "Back"
                        )
                    }
                },
                actions = {
                    IconButton(onClick = { viewModel.openBottomSheet(QuickActionSheetType.BILL) }) {
                        Icon(
                            imageVector = Icons.Default.Add,
                            contentDescription = if (isArabic) "إضافة فاتورة" else "Add Bill"
                        )
                    }
                }
            )
        }
    ) { paddingValues ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues),
            contentPadding = PaddingValues(16.dp)
        ) {
            items(bills) { bill ->
                Card(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 6.dp),
                    shape = RoundedCornerShape(DesignTokens.RadiusMedium),
                    colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surfaceVariant)
                ) {
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(16.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text(bill.title, style = MaterialTheme.typography.titleMedium.copy(fontWeight = FontWeight.Bold))
                            Text("Due: ${dateFormat.format(Date(bill.nextDueDate))} • ${bill.frequency}", style = MaterialTheme.typography.labelSmall.copy(color = MaterialTheme.colorScheme.onSurfaceVariant))
                            if (bill.isAutoPay) {
                                Text("⚡ Auto-Pay Enabled", style = MaterialTheme.typography.labelSmall.copy(color = MaterialTheme.colorScheme.primary))
                            }
                        }

                        Column(horizontalAlignment = Alignment.End) {
                            Text(
                                text = CurrencyFormatter.format(bill.amount, bill.currency),
                                style = MaterialTheme.typography.titleMedium.copy(fontWeight = FontWeight.Bold)
                            )
                            Spacer(modifier = Modifier.height(6.dp))
                            if (bill.status == BillStatus.PAID) {
                                Row(verticalAlignment = Alignment.CenterVertically) {
                                    Icon(Icons.Default.CheckCircle, contentDescription = null, tint = GreenIncome, modifier = Modifier.size(16.dp))
                                    Spacer(modifier = Modifier.width(4.dp))
                                    Text("Paid", style = MaterialTheme.typography.labelSmall.copy(color = GreenIncome, fontWeight = FontWeight.Bold))
                                }
                            } else {
                                Button(
                                    onClick = {
                                        val defaultAcc = accounts.firstOrNull()?.id ?: ""
                                        viewModel.payBill(bill, defaultAcc)
                                    },
                                    contentPadding = PaddingValues(horizontal = 12.dp, vertical = 4.dp)
                                ) {
                                    Text("Pay Now")
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
